qiskit/quantum_info/operators/symplectic/sparse_pauli_op.py

```diff
@@ -404,6 +404,46 @@ def simplify(self, atol=None, rtol=None):
             x = self.paulis.x[non_zero_indexes]
             z = self.paulis.z[non_zero_indexes]
             coeffs = coeffs[non_zero]
+
+        return SparsePauliOp(
+            PauliList.from_symplectic(z, x), coeffs, ignore_pauli_phase=True, copy=False
+        )
+
+    def chop(self, tol=1e-14):
+        """Remove real and imaginary parts of the coefficients that are close to 0.
+
+        For example, the operator representing ``1+1e-17j X + 1e-17 Y`` with a tolerance larger
+        than ``1e-17`` will be reduced to ``1 X`` whereas :meth:`.SparsePauliOp.simplify` would
+        return ``1+1e-17j X``.
+
+        Args:
+            tol (float): The absolute tolerance to check whether a real or imaginary part is 0.
+
+        Returns:
+            SparsePauliOp: This operator with chopped coefficients.
+        """
+        # Remove real (resp. imaginary) parts that are close to 0. This part for instance
+        # truncates 1+1e-17j to 1.0. Note that all coefficients that are left at this point
+        # are guaranteed to have a value above tolerance due to the previous is_zero check.
+        realpart_nonzero = np.logical_not(np.isclose(self.coeffs.real, 0, atol=tol))
+        imagpart_nonzero = np.logical_not(np.isclose(self.coeffs.imag, 0, atol=tol))
+
+        remaining_indices = np.logical_or(realpart_nonzero, imagpart_nonzero)
+        remaining_real = realpart_nonzero[remaining_indices]
+        remaining_imag = imagpart_nonzero[remaining_indices]
+
+        if not np.any(remaining_indices):
+            x = np.zeros((1, self.num_qubits), dtype=bool)
+            z = np.zeros((1, self.num_qubits), dtype=bool)
+            coeffs = np.array([0j], dtype=complex)
+        else:
+            coeffs = np.zeros(sum(remaining_indices), dtype=complex)
+            coeffs[remaining_real] += self.coeffs.real[realpart_nonzero]
+            coeffs[remaining_imag] += 1j * self.coeffs.imag[imagpart_nonzero]
```

Comment:
You should be able to assign directly to `coeffs.real` and `coeffs.imag`, which will be quite a bit more efficient because it limits you to real floating-point maths, and saves a vectorised complex multiplication.